### PR TITLE
700 may use problem json as default response object

### LINKED ISSUE #700: New Check for Zalando Rule [151]: MAY use ProblemJson as default response schema
Please check if the PR https://github.com/zalando/restful-api-guidelines/pull/369 introduces changes which are relevant to the Zally project.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #700

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/rule/zalando/JsonProblemAsDefaultResponseRule.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            .filterNot { "default" in it.second.keys }
+            .map { context.violation("operation has to contain the default response", it.first) }
+
+    @Check(severity = Severity.MAY)
+    fun checkDefaultResponseIsProblemJson(context: Context): List<Violation> = responsesPerOperation(context)
+            .filter { "default" in it.second.keys }
+            .filterNot { it.second["default"]!!.`$ref` in validRefs }
+            .map { context.violation("problem json has to be used as default response (${validRefs[0]})", it.first) }
+
+    private fun responsesPerOperation(context: Context): Collection<Pair<Operation, Map<String, ApiResponse>>> =
```
Comment (ddufourboivin-zalando) [inline server/src/main/java/de/zalando/zally/rule/zalando/JsonProblemAsDefaultResponseRule.kt]: Formatting of this method is strange. Can you auto-format in IJ and see what it gives us?

**Comment by ddufourboivin-zalando** [comment]
@maxim-tschumak : Is removing `NotSpecifyStandardErrorCodesRule` linked to adding the new rule?

**Comment by maxim-tschumak** [comment]
@ddufourboivin-zalando Yes, the old check was only a `HINT` (related to the same rule) which produced a lot of false-positives - I've decided to remove it.

**Comment by ddufourboivin-zalando** [comment]
:

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**